pkg/module_manager/models/modules/helm.go

```diff
@@ -0,0 +1,344 @@
+	"github.com/flant/addon-operator/pkg/helm/client"
+	"github.com/flant/addon-operator/pkg/utils"
+	"github.com/flant/kube-client/manifest"
+	"github.com/flant/shell-operator/pkg/utils/measure"
+)
+
+// HelmModule representation of the module, which has Helm Chart and could be installed with the helm lib
+type HelmModule struct {
+	// Name of the module
+	name string
+	// Path of the module on the fs
+	path string
+
+	values utils.Values
+
+	tmpDir string
+
+	dependencies *HelmModuleDependencies
+	validator    HelmValuesValidator
+}
+
+type HelmValuesValidator interface {
+	ValidateModuleHelmValues(string, utils.Values) error
+}
+
+type HelmResourceManager interface {
+	GetAbsentResources(manifests []manifest.Manifest, defaultNamespace string) ([]manifest.Manifest, error)
+	StartMonitor(moduleName string, manifests []manifest.Manifest, defaultNamespace string)
+	HasMonitor(moduleName string) bool
+}
+
+type MetricsStorage interface {
+	HistogramObserve(metric string, value float64, labels map[string]string, buckets []float64)
+}
+
+type HelmModuleDependencies struct {
+	HelmClientFactory   *helm.ClientFactory
+	HelmResourceManager HelmResourceManager
+	MetricsStorage      MetricsStorage
+	HelmValuesValidator HelmValuesValidator
+}
+
+// NewHelmModule build HelmModule from the Module templates and values + global values
+func NewHelmModule(bm *BasicModule, tmpDir string, deps *HelmModuleDependencies, validator HelmValuesValidator) (*HelmModule, error) {
+	moduleValues := bm.GetValues(false)
+
+	chartValues := map[string]interface{}{
+		"global": bm.dc.GlobalValuesGetter.GetValues(false),
+		utils.ModuleNameToValuesKey(bm.GetName()): moduleValues,
+	}
+
+	hm := &HelmModule{
+		name:         bm.Name,
+		path:         bm.Path,
+		values:       chartValues,
+		tmpDir:       tmpDir,
+		dependencies: deps,
+		validator:    validator,
+	}
+
+	isHelm, err := hm.isHelmChart()
+	if err != nil {
+		return nil, fmt.Errorf("create HelmModule failed: %w", err)
+	}
+
+	if !isHelm {
+		log.Infof("module %q has neither Chart.yaml nor templates/ dir, is't not a helm chart", bm.Name)
+		return nil, nil
+	}
+
+	return hm, nil
+}
+
+// isHelmChart check, could it be considered as helm chart or not
+func (hm *HelmModule) isHelmChart() (bool, error) {
+	chartPath := filepath.Join(hm.path, "Chart.yaml")
+
+	_, err := os.Stat(chartPath)
+	if err == nil {
+		// Chart.yaml exists, consider this module as helm chart
+		return true, nil
+	}
+
+	if os.IsNotExist(err) {
+		// Chart.yaml does not exist
+		// check that templates/ dir exists
+		_, err = os.Stat(filepath.Join(hm.path, "templates"))
+		if err == nil {
+			return true, hm.createChartYaml(chartPath)
+		}
+		if err != nil && os.IsNotExist(err) {
+			// if templates not exists - it's not a helm module
+			return false, nil
+		}
+	}
+
+	return false, err
+}
+
+func (hm *HelmModule) createChartYaml(chartPath string) error {
+	data := fmt.Sprintf(`name: %s
+version: 0.2.0`, hm.name)
```

Comment:
Well, honestly I wanted something like `0.0.1`, but we already have versions like [0.1.0](https://github.com/deckhouse/deckhouse/blob/main/modules/002-deckhouse/Chart.yaml#L2) or 0.1.1

I am confused with helm. It can abandon the upgrade because of lower version. And 1.0.0 looks redundant here. That why I decided like that: new minor version of addon-operator - new minor version of these generated helm charts

Reply:
Let's leave a comment here with the thing you described above.

Reply:
Done